packages/dds/tree/docs/roadmap.md

```diff
@@ -1,174 +1,399 @@
+This milestone makes it possible to read and write data to the Shared Tree without creating (reifing) that in-memory JavaScript representation.
+This is particularly useful in scenarios where the client has memory constraints or wants to maintains a copy of the data on the other side of an interop boundary (e.g., WASM, C++).
+It also allows clients/microservices to check permissions without loading the document and inspect changes without caring about the entire tree.
 
-#### Deliverables
+To accomplish this, the underlying Shared Tree layer is built on a [cursor API](https://github.com/microsoft/FluidFramework/blob/main/packages/dds/tree/src/core/tree/cursor.ts) that allows navigation of the tree by moving from node to node via explicit directional calls.
+Layers built on cursors are also able to remain agnostic to the structure of the tree it is navigating, allowing for flexible/multiple implementations.
+This cursor API is intended to be an expert API as working with it is more cumbersome compared with the more ergonomic APIs exposed in future milestones.
 
--   Changeset:
-    -   Encode constraints
-    -   Encode type info
-    -   Encode semantic info (hierarchical edits)
--   Constraints
-    -   'Subtree unchanged'
--   Schema
-    -   Parser
-    -   Repository
--   DDS
-    -   Conflict management
-        -   Detection
-        -   Revert
-        -   Handler
+While this is an important architectural milestone to build in early, the benefits around reification will not be fully realized until the _Storage performance: incrementality and virtualization_ milestone, as downloading the entire tree on load is currently required.
 
-### M4+: API and Modeling
+## Synchronous non-overlapping transactions
 
-M4 expands the audience for SharedTree by introducing higher level abstractions.
+> Complete
 
-### M5+: History and Branching
+Developers often want to group changes to the Shared Tree into logical units.
+Reasons for doing so include:
 
-M5 enables applications to create and merge branches from arbitrary points in history.
+-   Atomicity: a set of changes should be applied without any other changes (local or remote) being interleaved with them.
+-   App semantics: a set of changes represents a logical edit in the application model that must be applied atomically, and tree-level operations (such as undo/redo or history) should never result in an intermediate state being exposed.
+-   Dependencies between changes: changes within a grouping depend on some invariant (e.g., an insert should be applied only if none of the other changes in its group fail to apply due to concurrent edits).
 
-### M6+: Partial Checkout
+This milestone enables the creation of a single synchronous transaction per Shared Tree and guarantees atomicity (including for remote recipients of the edit).
+The other requirements will be satisfied by future milestones such as _Undo/redo_ and _Constraints_.
```

Comment:
It's desirable to ensure that people don't get the wrong impression when they skim headers and the associated statuses. It seems we're not ensuring that's the case here: the status flag says "Complete" but then we backpedal on that at the end by saying "actually, we're not there yet". The previous section about "Tree reading and writing without reification" seems to have a similar problem.

There a several ways we could address this:
1. Be more specific in the header so people get the right ideas as to what it is that is (for real) completed. 
2. Keep the simple header but add a "Part X/Y" to it and explain in the body what that part is.
3. Change the status flag to "in progress" or something like that.
4. Within that same header, break down the feature into parts and have a status flag for each part.

I think #1 would end up being cumbersome here. #3 is easy to do but information-poor. #4 makes the doc less linear in time, which is probably not what you want for a roadmap. So #2 would have my preference here.